--- a/Makefile.orig	2019-03-18 22:19:09.334942223 +0000
+++ b/Makefile	2019-03-18 22:22:34.800984111 +0000
@@ -57,7 +57,7 @@
 VERSION_REVISION=2
 VERSION_AGE=14
 
-PREFIX=/usr/local
+PREFIX ?= /usr/local
 LIBDIR=$(PREFIX)/lib
 INCDIR=$(PREFIX)/include
 MANDIR=$(PREFIX)/share/man
@@ -71,6 +71,8 @@
 
 $(LIBRARY): $(OBJECTS)
 	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath $(LIBDIR) -version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) $(LDFLAGS) -o $@ $^
+	ccache $(CC) -shared  -fPIC -DPIC  .libs/termkey.o .libs/driver-csi.o .libs/driver-ti.o   -ltinfo    -Wl,-soname -Wl,libtermkey.so.1 -o .libs/libtermkey.so.1.14.2
+
 
 demo: $(LIBRARY) demo.lo
 	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
