--- /dev/null	2019-03-07 11:00:10.312000000 +0000
+++ b/Makefile.am	2019-03-16 22:41:40.600357096 +0000
@@ -0,0 +1,11 @@
+bin_PROGRAMS = telepi
+
+telepi_SOURCES = telepi.c encode.c 
+
+telepi_LDADD = $(LIBOBJS) -lvchostif -lvcilcs -lopenmaxil -lilclient -lbcm_host -lvcos -lvchiq_arm -lpthread 
+telepi_LDFLAGS = -DHAVE_LIBBCM_HOST -DUSE_EXTERNAL_LIBBCM_HOST -DUSE_VCHIQ_ARM -Wno-psabi -DOMX_SKIP64BIT -lpthread
+telepi_CFLAGS = -DUSE_VCHIQ_ARM -DOMX_SKIP64BIT
+
+
+AM_CPPFLAGS = -I$(top_srcdir) -I$(STAGING_DIR)/usr/src/hello_pi/libs/ilclient/
+
--- /dev/null	2019-03-07 11:00:10.312000000 +0000
+++ b/configure.ac	2019-03-16 22:02:04.018115330 +0000
@@ -0,0 +1,13 @@
+AC_INIT([telepi], [1.0])
+AM_INIT_AUTOMAKE([foreign])
+AC_PROG_CC
+
+
+AC_CONFIG_FILES([Makefile])
+
+PKG_CHECK_MODULES([bcm_host], [bcm_host >= 1])
+
+AC_SUBST([TELEPI_CFLAGS])
+AC_SUBST([TELEPI_LIBS])
+AC_OUTPUT
+
