--- a/modules/phone-bluetooth/phone-bluetooth.pro.orig	2018-09-02 20:29:58.000000000 +0000
+++ b/modules/phone-bluetooth/phone-bluetooth.pro	2019-03-15 20:09:50.176863124 +0000
@@ -1,13 +1,12 @@
 TEMPLATE = lib
 CONFIG += c++11 plugin link_pkgconfig
-QT += qml quick bluetooth dbus BluezQt
+QT += qml quick bluetooth dbus
 TARGET = $$qtLibraryTarget(phone-bluetooth-plugin)
 DEFINES += QT_DEPRECATED_WARNINGS
 INCLUDEPATH += ../../includes
 DESTDIR = ../../plugins
-QMAKE_LIBS += -lKF5BluezQt
 
-PKGCONFIG += qofono-qt5
+PKGCONFIG += qofono-qt5 BluezQt
 
 SOURCES += \
     $$PWD/bluetooth.cpp \
