From 25f8da0a8979cdff2078d4f8c2d0fa2e64e542fb Mon Sep 17 00:00:00 2001
From: root <gregd72002@gmail.com>
Date: Mon, 1 Apr 2019 14:02:26 +0000
Subject: [PATCH] Run argument to whitelist plugins

---
 main.cpp          | 23 ++++++++++++++++++++++-
 pluginmanager.cpp | 10 +++++++++-
 pluginmanager.h   |  2 +-
 3 files changed, 32 insertions(+), 3 deletions(-)

diff --git a/main.cpp b/main.cpp
index 4de9ed4..b8cdd62 100644
--- a/main.cpp
+++ b/main.cpp
@@ -1,5 +1,6 @@
 #include <QApplication>
 #include <QQmlApplicationEngine>
+#include <QCommandLineParser>
 #include <QtQuick/QQuickView>
 #include <QtQml/QQmlContext>
 #include <QtQml/QQmlEngine>
@@ -34,8 +35,28 @@ int main(int argc, char *argv[])
 
     int defaultMenuItem = 4;
 
+
+    QCommandLineParser parser;
+    parser.setApplicationDescription("helper");
+    parser.addHelpOption();
+    parser.addVersionOption();
+    
+    QCommandLineOption pluginsOption(QStringList() << "p" << "plugins",
+	QCoreApplication::translate("main", "Plugins to enable (defaults to all)"),
+	QCoreApplication::translate("main", "plugins")
+    );
+    parser.addOption(pluginsOption);
+
+    parser.process(app);
+    QString p = parser.value(pluginsOption);
+    bool whitelist = parser.isSet(pluginsOption);
+    QStringList plugins;
+    if (whitelist) {
+	    plugins = p.split(" ",QString::SkipEmptyParts);
+    }
+   	 
     PluginManager pluginManager;
-    pluginManager.loadPlugins(engine);
+    pluginManager.loadPlugins(engine,whitelist,plugins);
 
     engine->rootContext()->setContextProperty("defaultMenuItem", defaultMenuItem);
 
diff --git a/pluginmanager.cpp b/pluginmanager.cpp
index a992bd0..73135eb 100644
--- a/pluginmanager.cpp
+++ b/pluginmanager.cpp
@@ -5,7 +5,7 @@ PluginManager::PluginManager(QObject *parent) : QObject(parent)
 
 }
 
-bool PluginManager::loadPlugins(QQmlApplicationEngine *engine)
+bool PluginManager::loadPlugins(QQmlApplicationEngine *engine, bool filter, QStringList filterList)
 {
     QDir pluginsDir(qApp->applicationDirPath());
 #if defined(Q_OS_WIN)
@@ -21,6 +21,14 @@ bool PluginManager::loadPlugins(QQmlApplicationEngine *engine)
     pluginsDir.cd("plugins");
     //Load plugins
     foreach (QString fileName, pluginsDir.entryList(QDir::Files)) {
+
+	QString fileBaseName=fileName.section(".",0,0);
+	if (filter && !filterList.contains(fileBaseName,Qt::CaseInsensitive)) {
+		qDebug() << "Plugin not whitelisted (disabled): " << fileBaseName;
+		continue;
+	}
+	qDebug() << "Loading plugin: " << fileName;
+
         QPluginLoader pluginLoader(pluginsDir.absoluteFilePath(fileName));
 
         if(pluginLoader.metaData().value("MetaData").type() != QJsonValue::Object){
diff --git a/pluginmanager.h b/pluginmanager.h
index 76be2c7..797be1f 100644
--- a/pluginmanager.h
+++ b/pluginmanager.h
@@ -20,7 +20,7 @@ class PluginManager : public QObject
 public:
     explicit PluginManager(QObject *parent = nullptr);
     ~PluginManager();
-    bool loadPlugins(QQmlApplicationEngine *engine);
+    bool loadPlugins(QQmlApplicationEngine *engine, bool filter, QStringList filterList);
 signals:
 
 public slots:
